internal/stanza/storage.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stanza
+
+import (
+	"context"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"go.opentelemetry.io/collector/component"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage"
+)
+
+func (r *receiver) setStorageClient(ctx context.Context, host component.Host) error {
+	// TODO define expected behavior if >1 storage extension found
```

Comment:
Probably fail. In the future we can introduce a config setting to select the storage if there is more than one.